File: app/src/main/java/io/neoterm/installer/packages/NeoPackageArchitecture.java
package io.neoterm.installer.packages;

/**
 * @author kiva
 */

public enum NeoPackageArchitecture {
    ALL, ARM, AARCH64, X86, X86_64;

    public static NeoPackageArchitecture parse(String arch) {
        switch (arch) {
            case "arm":
                return ARM;
            case "aarch64":
                return AARCH64;
            case "x86":
                return X86;
            case "x86_64":
                return X86_64;
            default:
                return ALL;
        }
    }
}

File: app/src/main/java/io/neoterm/customize/pm/NeoPackageArchitecture.kt
package io.neoterm.customize.pm

/**
 * @author kiva
 */

enum class NeoPackageArchitecture {
    ALL, ARM, AARCH64, X86, X86_64;


    companion object {

        fun parse(arch: String): NeoPackageArchitecture {
            when (arch) {
                "arm" -> return ARM
                "aarch64" -> return AARCH64
                "x86" -> return X86
                "x86_64" -> return X86_64
                else -> return ALL
            }
        }
    }
}
